include/class-polylang.php

```diff
@@ -239,37 +234,62 @@ public function init() {
 			 */
 			do_action( 'pll_model_init', $model );
 
-			$model->maybe_create_language_terms();
+			self::_init( $class, $model, __DIR__ );
+		}
+	}
 
-			/**
-			 * Fires after the $polylang object is created and before the API is loaded
-			 *
-			 * @since 2.0
-			 *
-			 * @param object $polylang
-			 */
-			do_action_ref_array( 'pll_pre_init', array( &$polylang ) );
+	/**
+	 * Polylang initialization.
+	 * Setups the Polylang Context, loads the modules and init Polylang.
+	 *
+	 * @since 3.6
+	 *
+	 * @param string    $class The class name.
+	 * @param PLL_Model $model Instance of PLL_Model.
+	 * @param string    $dir   The path to the include directory.
+	 * @return PLL_Base
+	 */
+	public static function _init( string $class, PLL_Model $model, string $dir ): PLL_Base {
+		global $polylang;
 
-			require_once __DIR__ . '/api.php'; // Loads the API
+		$links_model = $model->get_links_model();
 
-			// Loads the modules.
-			$load_scripts = glob( POLYLANG_DIR . '/modules/*/load.php', GLOB_NOSORT );
-			if ( is_array( $load_scripts ) ) {
-				foreach ( $load_scripts as $load_script ) {
-					require_once $load_script; // phpcs:ignore WordPressVIPMinimum.Files.IncludingFile.UsingVariable
-				}
-			}
+		/** @var PLL_Base $class */
+		$polylang = new $class( $links_model );
 
-			$polylang->init();
+		$model->maybe_create_language_terms();
 
-			/**
-			 * Fires after the $polylang object and the API is loaded
-			 *
-			 * @since 1.7
-			 *
-			 * @param object $polylang
-			 */
-			do_action_ref_array( 'pll_init', array( &$polylang ) );
+		/**
+		 * Fires after the $polylang object is created and before the API is loaded
+		 *
+		 * @since 2.0
+		 *
+		 * @param object $polylang
+		 */
+		do_action_ref_array( 'pll_pre_init', array( &$polylang ) );
+
+		// Loads the API
+		require_once $dir . '/api.php'; // phpcs:ignore WordPressVIPMinimum.Files.IncludingFile.UsingVariable
```

Comment:
Should we really ignore this CS warning?
☣As the `_init()` is public we can imagine someone who calls this method by passing a external path containing a malicious `api.php` file.
We need to find another way to pass the path as an unmutable string.

Reply:
Do you have any suggestions?

Maybe a public static method isn't the best idea either.